build/scripts/tests.sh

```diff
@@ -20,14 +35,47 @@ xunit_console_version="$(get_package_version xunitrunnerconsole)"
 dotnet_runtime_version="$(get_tool_version dotnetRuntime)"
 
 if [[ "${runtime}" == "dotnet" ]]; then
-    target_framework=netcoreapp2.1
     file_list=( "${unittest_dir}"/*/netcoreapp2.1/*.UnitTests.dll )
+    file_blacklist=(
+        # Disable the VB Semantic tests while we investigate the core dump issue
+        # https://github.com/dotnet/roslyn/issues/29660
+        "Microsoft.CodeAnalysis.VisualBasic.Semantic.UnitTests.dll"
+    )
     xunit_console="${nuget_dir}"/xunit.runner.console/"${xunit_console_version}"/tools/netcoreapp2.0/xunit.console.dll
-elif [[ "${runtime}" == "mono" ]]; then
-    file_list=(
-        "${unittest_dir}/Microsoft.CodeAnalysis.CSharp.Symbol.UnitTests/net46/Microsoft.CodeAnalysis.CSharp.Symbol.UnitTests.dll"
-        "${unittest_dir}/Microsoft.CodeAnalysis.CSharp.Syntax.UnitTests/net46/Microsoft.CodeAnalysis.CSharp.Syntax.UnitTests.dll"
-        )
+elif [[ "${runtime}" =~ ^(mono|mono-debug)$ ]]; then
+    file_list=( "${unittest_dir}"/*/net46/*.UnitTests.dll )
+    file_blacklist=(
+        'Microsoft.CodeAnalysis.CSharp.Scripting.UnitTests.dll'
+        # Missing mscoree.dll, other problems
+        'Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll'
+        # Omitted because we appear to be missing things necessary to compile vb.net.
+        # See https://github.com/mono/mono/issues/10679
+        'Microsoft.CodeAnalysis.VisualBasic.CommandLine.UnitTests.dll'
+        'Microsoft.CodeAnalysis.VisualBasic.Semantic.UnitTests.dll'
+        # PortablePdb and lots of other problems
+        'Microsoft.CodeAnalysis.VisualBasic.Scripting.UnitTests.dll'
+        # GetSystemInfo is missing, and other problems
+        # See https://github.com/mono/mono/issues/10678
+        'Microsoft.CodeAnalysis.CSharp.WinRT.UnitTests.dll'
+        # Many test failures
+        'Microsoft.CodeAnalysis.UnitTests.dll'
+        # Multiple test failures
+        'Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests.dll'
+        # Multiple test failures
+        'Microsoft.Build.Tasks.CodeAnalysis.UnitTests.dll'
+        # Various failures related to PDBs, along with a runtime crash
+        'Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll'
+        # Deadlocks or hangs for some reason
+        # FIXME: This is probably the new name for CompilerServer?
+        'VBCSCompiler.UnitTests.dll'
+        # Disabling on assumption
+        'Microsoft.CodeAnalysis.VisualBasic.Emit.UnitTests.dll'
+        # A zillion test failures + crash
+        # See https://github.com/mono/mono/issues/10756
+        'Microsoft.CodeAnalysis.VisualBasic.Symbol.UnitTests.dll'
+        # Deadlocks. 
+        'Microsoft.CodeAnalysis.CSharp.Symbol.UnitTests.dll'
```

Comment:
We're not seeing a deadlock here on stable or nightly. This has been running for quite some time now. Which version of Mono are you seeing the deadlock on? 

I'd like to keep this enabled if possible. It's been running in our CI for some time now.

Reply:
master branch mono from like a week ago. I'll remove it from the list if it will pass on roslyn CI. It is the thread abort deadlock you mentioned in slack, I thought?

Reply:
The deadlock I'm seeing is in the emit tests. It's not quite a deadlock though. xunit completes but won't exit. That usually means there is an unresolved `Task` that xunit is waiting for. 

In the past that happened in our tests which tested `Thread.Abort` logic. Hence I thought that might be the cause here. I think I've ruled that out now though. At the moment I have a script running locally to brute force figure out which test is hanging.